### PR TITLE
add precommit

### PR DESCRIPTION
**Pull request by callumrollo**
This draft PR adds some basic pre-commit hooks. This pre-commit will, among other things, satisfy the pyopensci requirement for automated linting

The linting is picking up quite a few PEP8 and spelling errors atm. These will need to be fixed. This also should be added to CI.

@rcaneill are you working on something similar already, or should I complete this PR?

### CODE REVIEW

**Comment by rcaneill** [comment]
HI @callumrollo ,
I am not working on the linting for xnemogcm, so I'm happy to get your contributions!

**Comment by callumrollo** [comment]
ok I think this is ready @rcaneill . Bit of a monster! Mostly spelling corrections and clearing out unused imports

File: xnemogcm/namelist.py
Comment (rcaneill) [inline xnemogcm/namelist.py:1]: It seems incorrect here to replace `nam` by `name` as `name` is already used in the for loop (line 27).

**Review by rcaneill — CHANGES_REQUESTED** [review]
Thanks for the great work! See my comment in file namelist.py

**Review by rcaneill — APPROVED** [review]

**Pull request merged**